### DISCUSSION TITLE
cmgen for Android

### DISCUSSION THREAD

**Discussion by CloseFile** [google/filament#6705] [Ideas]
Are there any plans to add this feature for converting on device ?

**Comment by romainguy** [comment]
You can use libiblprefilter for this.

**Reply by CloseFile** [reply]
Thank for your advice.
Works great with hdri from polyhaven. But how could display 360 panorama .png or .jpg as a skybox ?

**Reply by romainguy** [reply]
For this you should probably just create your own skybox that samples from an equirectangular image. The skybox is a single triangle that uses this material, for reference: https://github.com/google/filament/blob/main/filament/src/materials/skybox.mat

**Reply by CloseFile** [reply]
> For this you should probably just create your own skybox that samples from an equirectangular image. The skybox is a single triangle that uses this material, for reference: https://github.com/google/filament/blob/main/filament/src/materials/skybox.mat

it's too difficult for me without examples. Thanks anyway for your help
